### DISCUSSION TITLE
How to filter values from a dropdown field by taking into account value from another dropdown previously set

### DISCUSSION THREAD

**Discussion by RoKrypto** [wintercms/winter#658] [Q&A] [answered]
Greetings.

I hope someone can help me with the following.

I have a form to put prices to cars. Inside this form, I have a series of fields, from which, two are relation-typed dropdowns: models and versions. Each model can have many versions. As some versions might have the same name, I need to figure out a way to implement some functionality oriented to filter the Versions field by taking into account the value set previously for the Models field. Otherwise, I might assign the incorrect price to the incorrect version.

I read the documentation and perhaps what I need has to do with either Events or Field Dependency. But I have to be honest, I have no clue on how to proceed. Could someone kindly enlighten the way for me?.

**Comment by LukeTowers** [comment, marked as answer]
You're looking for Field Dependency and model option methods.

Example:

**fields.yaml**
```yaml
model:
    label: Model
    type: dropdown
version:
    label: Version
    type: dropdown
    dependsOn: model
```

**MyModel.php**
```php
public function getModelOptions()
{
    return ModelModel::lists('name', 'id');
}

public function getVersionOptions()
{
    if ($this->model) {
        return $this->model->versions()->lists('name', 'id');
    }
    return Version::lists('name', 'id');
}
```

**Reply by RoKrypto** [reply]
You're amazing, @LukeTowers 😀 Thank you so much for your support.

**Reply by LukeTowers** [reply]
For anyone else's future reference, see https://wintercms.com/docs/backend/forms#field-dependencies
